rustls/src/lib.rs

```diff
@@ -673,3 +673,20 @@ const TEST_PROVIDERS: &[&crypto::CryptoProvider] = &[
     #[cfg(feature = "ring")]
     &crypto::ring::DEFAULT_PROVIDER,
 ];
+
+mod core_hash_polyfill {
```

Comment:
Nit: maybe avoid the need to leak this module name into all the imports by re-"exporting" (`pub(crate)`) `DynHasher` from the crate root?